ui/src/pages/Home/Profile/Profile.tsx

```diff
@@ -36,24 +47,26 @@ export function Profile() {
         setLoading(false);
       }
     };
+
     fetchData();
   }, []);
+
   return (
     <div className="profile">
       {loading ? (
-        <>loading</>
+        <div className="profile__skeleton">
+          <div className="profile__skeleton__text">loading</div>
+        </div>
       ) : (
         <>
-          <div>
-            <img
-              className="profile__image"
-              src={`/images/${profileImages[RandomImage()]}`}
-              alt="bunnie"
-            />
-          </div>
-          <div className="profile__text">
-            {profile[0].bio}
-          </div>
+          <img
+            className={
+              isImageLoaded ? "profile__image" : "profile__image__loading"
+            }
```

Comment:
Should this also have a skeleton of its own?

Reply:
So how we have it is that once everything loads, including the actual text AND image does the loading go away, this is to prevent the text to just jump randomly: 
I.E. once the text loads but the image doesnt, the text loads on the top left, but once the image loads, the text jumps to the offset of the image, weird behavior and i dont like it. So this prevents that

Reply:
right makes sense